SwaggerGen/Swagger/Type/AllOfType.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace SwaggerGen\Swagger\Type;
+
+/**
+ * allOf compound type
+ *
+ * @package    SwaggerGen
+ * @author     Bruce Weirdan <weirdan@gmail.com>
+ * @copyright  2014-2015 Martijn van der Lee
+ * @license    https://opensource.org/licenses/MIT MIT
+ */
+class AllOfType extends AbstractType
+{
+	private $allOfItems = array();
+	private $mostRecentItem;
+	protected function parseDefinition($definition)
+	{
+		$pattern = self::REGEX_START . 'allOf' . self::REGEX_CONTENT . self::REGEX_END;
+		$inlineDef = '';
+		if (preg_match($pattern, $definition, $matches)) {
+			if (isset($matches[1])) {
+				$inlineDef = $matches[1];
+			}
+		}
+		if ($inlineDef) {
+			foreach ($this->parseList($inlineDef) as $item) {
+				$this->handleCommand('item', $item);
+			}
+		}
+	}
+
+	/**
+	 * @param string $list
+	 * @return string
+	 * @todo remove duplication between this method and ObjectType::extract_property()
```

Comment:
A protected static method in `AbstractType` (like `swap` method) would be the most obvious place to merge+refactor